app/code/core/Mage/Usa/Model/Shipping/Carrier/Ups.php

```diff
@@ -1201,7 +1201,7 @@ protected function _parseRestTrackingResponse($trackingValue, $jsonResponse)
         if ($jsonResponse) {
             $responseData = json_decode($jsonResponse, true);
 
-            if ($responseData['trackResponse']['shipment']) {
+            if (isset($responseData['trackResponse']['shipment'])) {
```

Comment:
This change solves https://github.com/OpenMage/magento-lts/pull/3878#issuecomment-2098924102 but the array structure still needs to be verified

Reply:
> but the array structure still needs to be verified

I think what you have here is enough. If this is a success (200) response from the API, the key(s) for response.trackResponse.shipment are, I think, always present. If they aren't there, that implies we got a non-happy response from UPS.

Reply:
for the error it needed to be $responseData['response'], for some reason, that's why I put this note ;-)